check-oracle.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# used for oracle ORAchk.  install, run and uninstall ORAchk
+#
+# Check if we're using the Mac stock getopt and fail if true
+# shellcheck disable=SC2034
+out="$(getopt -T)"
+if [ $? != 4 ]; then
+    echo -e "Your getopt does not support long parameters, possibly you're on a Mac, if so please install gnu-getopt with brew"
+    echo -e "\thttps://brewformulas.org/Gnu-getopt"
+    exit 1
+fi
+
+# set to 'echo ' to only dislay commands
+DEBUG_CMD='echo '
+DEBUG_CMD=''
+
+ORA_SWLIB_BUCKET_PARAM='^gs://.+[^/]$'
+
+ORACLE_SERVER=''
+AHF_DIR='AHF'
+AHF_FILE=''
+AHF_INSTALL=0
+AHF_UNINSTALL=0
+RUN_ORACHK=0
+
+GETOPT_MANDATORY="oracle-server:,oracle-sid:"
+GETOPT_OPTIONAL="ora-swlib-bucket:,ahf-file:,ahf-dir:,ahf-install,ahf-uninstall,run-orachk,help"
+
+GETOPT_LONG="$GETOPT_MANDATORY,$GETOPT_OPTIONAL"
+GETOPT_SHORT="h"
+
+options="$(getopt --longoptions "$GETOPT_LONG" --options "$GETOPT_SHORT" -- "$@")"
+
+# shellcheck disable=SC2181
+[ $? -eq 0 ] || {
+    echo "Invalid options provided: $*" >&2
+    exit 1
+}
+
+eval set -- "$options"
+
+help () {
+    echo -e "\tUsage: $(basename "$0")"
+    echo "${GETOPT_MANDATORY}" | sed 's/,/\n/g' | sed 's/:/ <value>/' | sed 's/\(.\+\)/\t --\1/'
+    echo "${GETOPT_OPTIONAL}"  | sed 's/,/\n/g' | sed 's/:/ <value>/' | sed 's/\(.\+\)/\t [ --\1 ]/'
+    echo
+    echo "--ahf-install and --run-orachk may be combined to install and run"
+
+}
+
+# check if both install and run were specified together
+RUN_ENABLED=0
+INSTALL_ENABLED=0
+
+while true
+do
+
+    case "$1" in
+
+        --ora-swlib-bucket)
+           ORA_SWLIB_BUCKET="$2"
+           shift
+           ;;
+
+        --help | -h)
+            help >&2
+            exit 0
+            ;;
+
+        --ahf-file) 
+            AHF_FILE="$2"
+            shift
+            ;;
+ 
+        --ahf-dir)
+            AHF_DIR="$2"
+             shift
+             ;;
+
+        --oracle-sid)
+            ORACLE_SID="$2"
+            shift
+            ;;
+
+        --oracle-server)
+            ORACLE_SERVER="$2"
+            shift
+            ;;
+
+        --ahf-install)
```

Comment:
If no command is specified, this script exits with success and no output at all.  It would be better to show an error in this case.

Reply:
Something here does not seem right.

When I run without any options, I get this:

```text
$ ./check-oracle.sh
please specify --oracle-server
```

When run with `--ahf-install`

```text
$ ./check-oracle.sh --ahf-install
please specify --oracle-server
```

I think it would probably be good to include the help message here, but, there is output.

Can you verify that check-oracle.sh is correct?

What I see in google-toolkit-for-oracle in rel/oratk-57-orachk

```text
$ sha256sum check-oracle.sh
1ffdb2c72f52ce073f8a7b0c6cc3309653adc9226cef072a4faf06174f9a0928  check-oracle.sh
```

Reply:
In my case, I'm saw the error message and passed `--oracle-server` as requested (but no command).  And I got success and blank output.